modules/service/src/main/resources/db/migration/V1001__more_obscalc_triggers.sql

```diff
@@ -0,0 +1,98 @@
+-- Invalidate the obscalc results for all observations in the program
+CREATE OR REPLACE PROCEDURE invalidate_all_obscalc_for_program(
+  pid d_program_id
+)
+  LANGUAGE plpgsql AS $$
+DECLARE
+  obs_id d_observation_id;
+BEGIN
+  FOR obs_id IN
+    SELECT c_observation_id
+    FROM t_observation
+    WHERE c_program_id = pid
```

Comment:
I suppose we could skip deleted observations.  If they are undeleted then it's an obs edit a calc is triggered for that.